pillar/components/haproxy.sls

```diff
@@ -27,3 +27,4 @@ haproxy:
         s3server:
             ssl_enabled: true
     nbproc: 12
+    nbthread: 16
```

Comment:
Get the nbthread value through if/else according to HW/VM
Put nbthread value as 32 for HW and 2 for VM